pulsar-broker/src/main/java/org/apache/pulsar/broker/service/PulsarMetadataEventSynchronizer.java

```diff
@@ -114,62 +133,100 @@ private void publishAsync(MetadataEvent event, CompletableFuture<Void> future) {
-                                            .collect(Collectors.toList()))
-                                    .thenApply(__ -> c.acknowledgeAsync(msg)).exceptionally(ex -> {
-                                        log.warn("Failed to synchronize {} for {}", msg.getMessageId(), topicName);
-                                        return null;
-                                    });
-                        }
-                    } catch (Exception e) {
-                        log.warn("Failed to synchronize {} for {}", msg.getMessageId(), topicName);
+            .topic(topicName).subscriptionName(SUBSCRIPTION_NAME).ackTimeout(60, TimeUnit.SECONDS)
+            .subscriptionType(SubscriptionType.Failover).messageListener((c, msg) -> {
+                log.info("Processing metadata event for {} with listeners {}", msg.getValue().getPath(),
+                        listeners.size());
+                try {
+                    if (listeners.size() == 0) {
+                        c.acknowledgeAsync(msg);
+                        return;
+
                     }
-                });
+                    if (listeners.size() == 1) {
+                        listeners.get(0).apply(msg.getValue()).thenApply(__ -> c.acknowledgeAsync(msg))
+                                .exceptionally(ex -> {
+                                    log.warn("Failed to synchronize {} for {}", msg.getMessageId(), topicName,
+                                            ex.getCause());
+                                    return null;
+                                });
+                    } else {
+                        FutureUtil
+                                .waitForAll(listeners.stream().map(listener -> listener.apply(msg.getValue()))
+                                        .collect(Collectors.toList()))
+                                .thenApply(__ -> c.acknowledgeAsync(msg)).exceptionally(ex -> {
+                                    log.warn("Failed to synchronize {} for {}", msg.getMessageId(), topicName);
+                                    return null;
+                                });
+                    }
+                } catch (Exception e) {
+                    log.warn("Failed to synchronize {} for {}", msg.getMessageId(), topicName);
+                }
+            });
         consumerBuilder.subscribeAsync().thenAccept(consumer -> {
-            log.info("successfully created consumer {}", topicName);
-            this.consumer = consumer;
+            backOff.reset();
+            if (STATE_UPDATER.compareAndSet(this, State.Starting_Consumer, State.Started)) {
+                this.consumer = consumer;
+                log.info("successfully created consumer {}", topicName);
+            } else {
+                State stateTransient = state;
+                log.info("[{}] Closing the new consumer because the synchronizer state is {}", stateTransient);
```

Comment:
Fewer arguments provided (1) than placeholders specified (2)

Reply:
Fixed it in the next PR : https://github.com/apache/pulsar/pull/22695